src/marten/db/field/slug.cr

```diff
@@ -14,11 +20,27 @@ module Marten
           @null = false,
           @unique = false,
           @index = true,
-          @db_column = nil
+          @db_column = nil,
+          @slugify : Symbol? = nil,
+          @slugify_cb : (::String -> ::String) = ->(value : ::String) { generate_slug(value) }
         )
+          if @slugify
+            @null = true
+            @blank = true
+          end
+        end
+
+        macro check_definition(field_id, kwargs)
+          # No-op max_size automatic checks...
         end
 
         def validate(record, value)
+          if slugify?(value)
+            slug = slugify_cb.call(record.get_field_value(slugify.not_nil!).to_s)
+            record.set_field_value(id, slug)
+            return
```

Comment:
IMHO we shouldn't return early here because we will skip other validations by doing so.